lib/puppet/provider/exec.rb

```diff
@@ -41,6 +41,13 @@ def run(command, check = false)
                 warning "Overriding environment setting '#{env_name}' with '#{value}'"
               end
               environment[env_name] = value
+            elsif setting =~ /^(\w+)=$/
+              env_name = $1
+              warning "Environment Setting #{env_name} set to `#{env_name}=`, if this is unexpected check the variable has been assigned in the string"
```

Comment:
Interesting that it always warns when setting env variables. I'd like to see the messages cleaned up a bit so we always produce one warning that describes whether we're unsetting an environment variable or having no effect. Especially since the 2nd warning right now will always have an empty-string value.